frontend/src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState } from 'react';
+import { fetchSuggestions, Suggestion } from '../services/textureService';
+import InputSection from './InputSection';
+import SuggestionList from './SuggestionList';
+import styles from './Autocomplete.module.css';
+
+interface AutocompleteProps {
+  onTextureSelection: (texture: Suggestion) => void;
+  cardRemoved: boolean;
+}
+
+const Autocomplete: React.FC<AutocompleteProps> = ({
+  onTextureSelection,
+  cardRemoved,
+}) => {
+  const [input, setInput] = useState('');
+  const [suggestions, setSuggestions] = useState<Suggestion[]>([]);
+  const [selectedSuggestion, setSelectedSuggestion] = useState(-1);
+  const [limit, setLimit] = useState(5);
+  const [showSuggestions, setShowSuggestions] = useState(true);
+  const [error, setError] = useState(false);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInput(e.target.value);
+    setShowSuggestions(true);
+  };
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    const key = e.key;
+    switch (key) {
+      case 'ArrowUp':
+        if (selectedSuggestion > 0) {
+          setSelectedSuggestion(selectedSuggestion - 1);
+        }
+        break;
+      case 'ArrowDown':
+        if (selectedSuggestion < suggestions.length - 1) {
+          setSelectedSuggestion(selectedSuggestion + 1);
+        }
+        break;
+      case 'Enter':
+        if (
+          selectedSuggestion >= 0 &&
+          selectedSuggestion < suggestions.length
+        ) {
+          setInput(suggestions[selectedSuggestion].name);
+          setSuggestions([]);
+          setSelectedSuggestion(-1);
+          setShowSuggestions(false);
+          onTextureSelection(suggestions[selectedSuggestion]);
+        }
+        break;
+    }
+  };
+
+  const handleLimitChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const newLimit = parseInt(e.target.value, 10);
+    setLimit(newLimit);
+    setError(newLimit < 1 || newLimit > 5);
+  };
+
+  const handleSuggestionClick = (suggestion: Suggestion) => {
+    setInput(suggestion.name);
+    setSuggestions([]);
+    setSelectedSuggestion(-1);
+    setShowSuggestions(false);
+    onTextureSelection(suggestion);
+  };
+
+  const handleClickOutside = (e: MouseEvent) => {
+    if (
+      e.target instanceof Element &&
+      !e.target.closest(`.${styles.autocomplete}`)
+    ) {
+      setShowSuggestions(false);
+    }
+  };
+
+  React.useEffect(() => {
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, []);
+
+  React.useEffect(() => {
+    const fetchSuggestionsAsync = async () => {
```

Comment:
The `fetchSuggestionsAsync` method could be refactored out of the useEffect hook into a useCallback hook to make it a bit more reusable and readable. It would look a bit like this:

![Screenshot 2023-04-18 at 1 38 15 PM](https://user-images.githubusercontent.com/8660478/232859163-8b3bb5de-6973-4a35-8ae1-67e37fa4745e.png)

Notice the self-calling function here `(async () => { await fetchSuggestionsAsync() })();` which is a little trick to properly call async functions inside a useEffect hook.